src/test/java/uk/ac/cam/cl/dtg/segue/api/managers/GroupManagerTest.java

```diff
@@ -123,4 +135,33 @@ public final void groupManager_createValidGroup_aGroupShouldBeCreated() {
     }
     verify(this.groupDataManager);
   }
+
+  @Test
+  public void orderUsersByName_ordersBySurnamePrimarily() {
+    List<RegisteredUserDTO> users = Stream.of(
+            new RegisteredUserDTO("A", "Ab", "aab@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.MALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO("B", "Ar", "bar@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.FEMALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO("C", "Ax", "caz@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.MALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO(null, "Ax", "NONEax@test.com", EmailVerificationStatus.VERIFIED, somePastDate,
+                Gender.FEMALE, somePastDate, "", false),
+            new RegisteredUserDTO("A", "Ba", "dba@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.FEMALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO("B", "Bb", "ebb@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.MALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO("C", "Bf", "fbf@test.com", EmailVerificationStatus.VERIFIED, somePastDate, Gender.FEMALE,
+                somePastDate, "", false),
+            new RegisteredUserDTO("A", null, "aNONE@test.com", EmailVerificationStatus.VERIFIED, somePastDate,
+                Gender.FEMALE, somePastDate, "", false))
+        .peek(user -> user.setId((long) ("" + user.getGivenName() + user.getFamilyName()).hashCode()))
```

Comment:
Was this empty string intended?

Reply:
I copied the test directly from one file to the other when unifying them, that empty string was also in the original. Only bit I changed was replacing the whitebox invocation

Reply:
I cannot think of any good reason, to be honest. I think if you remove it, the test should still behave in the same way.